### ISSUE TITLE
Analytics is not started on 7.5.0 and later versions

### ISSUE THREAD

**Issue by fthdgn** [firebase/firebase-ios-sdk#7595]
### [REQUIRED] Step 1: Describe your environment

  * Xcode version:12.4 (12D4e)
  * Firebase SDK version: 7.5.0, 7.6.0, 7.7.0
  * Installation method: `CocoaPods`
  * Firebase Component: Analytics

#### Steps to reproduce:

The latest version of our app is not visible on Firebase console. A/B testing is not working, and Latest Release does not detect the release even though, according to Apple Analytics, the sessions of newest release surpassed the sessions of old version.
When we analyzed console logs of previous version and the latest version, we noticed that Firebase Analytics are does not start on the latest version. `Analytics v.7.6.0 started` line is not printed. We tested 7.5.0 and 7.7.0, the line is not printed on them too. The line is printed on 7.4.0. The line also printed on Debug build. The problem occurs on Release build since 7.5.0.
We are suspecting the reason of Firebase console problems is related the lack of `Analytics v.7.7.0 started` log.

We cannot find anything on changelog and documentations.

#### Relevant Code:

```
    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
        ...
        FirebaseApp.configure()
        ...
        return true
    }
 ```

**Comment by paulb777** [comment]
I've confirmed that I see `Analytics v.7.7.0 started` with the latest version installed via CocoaPods.

I suspect something is corrupt in your Xcode workspace. Perhaps `pod deintegrate` and `pod update` would resolve?

**Comment by paulb777** [comment]
Or maybe an issue like #7521?

**Comment by morganchen12** [comment]
Also, check to make sure you don't have duplicate Firebase class definitions in your app's runtime.

**Comment by fthdgn** [comment]
`pod deintegrate` and `pod update` didn't solve the problem.

But adding `-ObjC` option to `Other Linker Flags` solved the problem, `Analytics v.7.7.0 started` is printed on Release build.
CocoaPods adds `-ObjC` itself on 7.4.0, but not on 7.7.0

SPM documentation mentions this, but it was not required for CocoaPods.
Is it required now?

**Comment by paulb777** [comment]
Interesting. It's likely a side-effect of a [CocoaPods bug](https://github.com/CocoaPods/CocoaPods/pull/10234) that was fixed in CocoaPods 1.10.1 that we added a workaround for in Firebase 7.5.0.

I suspect your use case may be something that we didn't test. Are you calling using Analytics APIs in the app? Or are you completely relying on Analytics being implicitly linked?

Either way, I suspect updating to CocoaPods 1.10.1 would be an alternative to needing to manually add the `-ObjC` option.

cc: @htcgh

**Comment by fthdgn** [comment]
I created a project to reproduce problem.
7.7.0 does not add -ObjC and does not print log.
7.4.0 adds -ObjC and prints log.
Pod version is 1.10.1

There is no API call. Just `FirebaseApp.configure()` on `application(_:didFinishLaunchingWithOptions:)`

https://github.com/fthdgn/firebase-ios-sdk-issue-7595

**Comment by Nuckijfuta** [comment]
> I found a few problems with this issue:
>   * I couldn't figure out how to label this issue, so I've labeled it for a human to triage. Hang tight.
>   * This issue does not seem to follow the issue template. Make sure you provide all the required information.

Need to be able to make it work

**Comment by paulb777** [comment]
@fthdgn Thanks for the reproducible example. 👍  I've reproduced and will investigate.

**Comment by paulb777** [comment]
Open CocoaPods issue https://github.com/CocoaPods/CocoaPods/issues/10459

**Comment by paulb777** [comment]
Proposed CocoaPods fix at https://github.com/CocoaPods/CocoaPods/pull/10460

**Comment by paulb777** [comment]
I'm going to close this here - since the fix will likely be part of CocoaPods 1.10.2. Track at CocoaPods/CocoaPods#10460

In the meantime, adding the `-ObjC` option to the `Other Linker Flags` Build Setting is the right workaround.

Thanks to @fthdgn for raising the issue and the repro case.
